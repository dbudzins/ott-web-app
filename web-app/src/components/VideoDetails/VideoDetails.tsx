import React from 'react';
import classNames from 'classnames';

import styles from './VideoDetails.module.scss';

import CollapsibleText from '#src/components/CollapsibleText/CollapsibleText';
import useBreakpoint, { Breakpoint } from '#src/hooks/useBreakpoint';

type PosterMode = 'fading' | 'normal';

type Props = {
  title: string;
  description: string;
  primaryMetadata: React.ReactNode;
  secondaryMetadata?: React.ReactNode;
  poster?: string;
  posterMode: PosterMode;
  startWatchingButton: React.ReactNode;
  shareButton: React.ReactNode;
  favoriteButton: React.ReactNode;
  trailerButton: React.ReactNode;
  childrenPadding?: boolean;
};

const VideoDetails: React.FC<Props> = ({
  title,
  description,
  primaryMetadata,
  secondaryMetadata,
  poster,
  posterMode,
  children,
  startWatchingButton,
  shareButton,
  favoriteButton,
  trailerButton,
  childrenPadding = true,
}) => {
  const breakpoint: Breakpoint = useBreakpoint();
  const isMobile = breakpoint === Breakpoint.xs;

  return (
    <div className={styles.video} data-testid="video-details">
      <div
        className={classNames(styles.main, styles.mainPadding, {
          [styles.posterNormal]: poster === 'normal',
        })}
      >
        <div className={styles.info}>
          <h2 className={styles.title}>{title}</h2>
          <div className={styles.metaContainer}>
            <div className={styles.primaryMetadata}>{primaryMetadata}</div>
            {secondaryMetadata && <div className={styles.secondaryMetadata}>{secondaryMetadata}</div>}
          </div>
          <CollapsibleText text={description} className={styles.description} maxHeight={isMobile ? 60 : 'none'} />

          <div className={styles.buttonBar}>
            {startWatchingButton}
            {trailerButton}
            {favoriteButton}
            {shareButton}
          </div>
        </div>
        <div className={classNames(styles.poster, styles[posterMode])} style={{ backgroundImage: `url('${poster}')` }} />
      </div>
      {!!children && <div className={classNames(styles.related, { [styles.mainPadding]: childrenPadding })}>{children}</div>}
    </div>
  );
};

export default VideoDetails;
