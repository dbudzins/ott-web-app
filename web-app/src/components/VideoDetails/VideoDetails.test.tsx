import React from 'react';
import { render } from '@testing-library/react';

import VideoDetails from './VideoDetails';

describe('<VideoDetails>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(
      <VideoDetails
        title="Test video"
        description="Video description"
        primaryMetadata="Primary metadata string"
        secondaryMetadata={<strong>Secondary metadata string</strong>}
        poster="posterimage"
        posterMode="fading"
        startWatchingButton={<button>Start watching</button>}
        shareButton={<button>share</button>}
        favoriteButton={<button>favorite</button>}
        trailerButton={<button>play trailer</button>}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
