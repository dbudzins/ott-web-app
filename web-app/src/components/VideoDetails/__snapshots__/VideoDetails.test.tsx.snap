// Vitest Snapshot v1

exports[`<VideoDetails> > renders and matches snapshot 1`] = `
<div>
  <div
    class="video"
    data-testid="video-details"
  >
    <div
      class="main mainPadding"
    >
      <div
        class="info"
      >
        <h2
          class="title"
        >
          Test video
        </h2>
        <div
          class="metaContainer"
        >
          <div
            class="primaryMetadata"
          >
            Primary metadata string
          </div>
          <div
            class="secondaryMetadata"
          >
            <strong>
              Secondary metadata string
            </strong>
          </div>
        </div>
        <div
          class="collapsibleText"
        >
          <div
            class="textContainer description"
            style="max-height: none;"
          >
            Video description
          </div>
        </div>
        <div
          class="buttonBar"
        >
          <button>
            Start watching
          </button>
          <button>
            play trailer
          </button>
          <button>
            favorite
          </button>
          <button>
            share
          </button>
        </div>
      </div>
      <div
        class="poster fading"
        style="background-image: url(posterimage);"
      />
    </div>
  </div>
</div>
`;
